sources/metricdog/src/service_check.rs

```diff
@@ -0,0 +1,159 @@
+use crate::error::{self, Result};
+use log::trace;
+use snafu::ResultExt;
+use std::process::Command;
+
+#[derive(Debug, Clone, Eq, PartialEq, Ord, PartialOrd)]
+pub(crate) struct ServiceHealth {
+    /// Whether or not the service is healthy.
+    pub(crate) is_healthy: bool,
+    /// In the event of an unhealthy service, the service's exit code (if found).
+    pub(crate) exit_code: Option<i32>,
+}
+
+pub(crate) trait ServiceCheck {
+    /// Checks the given service to see if it is healthy.
+    fn check(&self, service_name: &str) -> Result<ServiceHealth>;
+}
+
+pub(crate) struct SystemdCheck {}
+
+impl ServiceCheck for SystemdCheck {
+    fn check(&self, service_name: &str) -> Result<ServiceHealth> {
+        if is_ok(service_name)? {
+            return Ok(ServiceHealth {
+                is_healthy: true,
+                exit_code: None,
+            });
+        }
+        Ok(ServiceHealth {
+            is_healthy: false,
+            exit_code: parse_service_exit_code(service_name)?,
+        })
+    }
+}
+
+struct Outcome {
+    exit: i32,
+    stdout: String,
+}
+
+impl Outcome {
+    fn is_exit_true(&self) -> bool {
```

Comment:
Out of curiosity, why "true" here?

Reply:
Two of the command-line calls use exit zero to represent true, so I guess I was trying to make the check for `== 0` to read more like a check for a truthy response.